libeos-updater-util/avahi-service-file.h

```diff
@@ -44,4 +44,241 @@ gboolean eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+ * cannot exceed the 250 bytes to fit it in the TXT record. The maths
+ * behind it is as follows:
+ *
+ * The TXT record can have maximum 256 bytes. 1 byte is reserved
+ * implicitly for the size of the record (you can think about the
+ * record as a pascal string). 2 bytes go for the name of the TXT
+ * record (it is "rb" from "refs bloom"). 1 byte go for the equal
+ * sign. 1 byte goes for the bloom k paramater and 1 byte goes for the
+ * bloom hashing function ID. That gives us 250 bytes max.
+ *
+ * Default value of this option (if not overridden) is 250.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U "bloom-size"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q:
+ *
+ * Specifies the repo index for which the service file will be
+ * generated.
+ *
+ * Default value of this option (if not overridden) is 0.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q "repository-index"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_PORT_Q:
+ *
+ * Specifies the port where the server serving the repository contents
+ * is listening.
+ *
+ * Default value of this option (if not overridden) is set at the
+ * compilation time.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_PORT_Q "port"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y:
+ *
+ * Specifies the size limit generated TXT records can have.
+ * Possible values:
+ *
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_SUPPORT_CRAPPY_HARDWARE
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_ETHERNET_PACKET
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_MULTICAST_DNS_PACKET
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_16_BIT_LIMIT
+ * - %EOS_OSTREE_AVAHI_SIZE_LEVEL_ABSOLUTELY_LAX
+ *
+ * Default value of this option (if not overridden) is
+ * %EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y "txt-records-size-level"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T:
+ *
+ * Specifies the custom size limit. Only applicable if
+ * %EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y was set to
+ * %EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM.
+ *
+ * It has no default value - must be specified explicitly.
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T "txt-records-custom-size"
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID:
+ *
+ * Hash ID that maps to the ostree_str_bloom_hash() function. It takes
+ * nul-terminated strings as an input.
+ */
+#define EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID ((guint8)1)
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM:
+ *
+ * A possible value for the
+ * %EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y option. It means
+ * that the size limit is specified in the
+ * %EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T option.
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM ((guint8)0)
```

Comment:
Uh, I think it was because enums are a bit underspecified (what size are they?), so they may need casting when passed to varargs functions like `g_variant_new`.